src/main/java/com/bobocode/bibernate/configuration/PersistenceUnitProperties.java

```diff
@@ -0,0 +1,77 @@
+package com.bobocode.bibernate.configuration;
+
+import static com.bobocode.bibernate.configuration.PropertyValues.DATA_SOURCE_JDBC_URL_PROPERTY;
+import static com.bobocode.bibernate.configuration.PropertyValues.DATA_SOURCE_PASSWORD_PROPERTY;
+import static com.bobocode.bibernate.configuration.PropertyValues.DATA_SOURCE_USER_PROPERTY;
+import static com.bobocode.bibernate.configuration.PropertyValues.DIALECT_PROPERTY;
+import static com.bobocode.bibernate.configuration.PropertyValues.PERSISTENCE_UNIT_NAME_PROPERTY;
+
+import com.bobocode.bibernate.exception.BibernateException;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Optional;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Getter
+@Slf4j
+public class PersistenceUnitProperties {
+
+    private String persistenceUnitName;
+    private String databaseName;
+    private Dialect dialect;
+    private String jdbcUrl;
+    private String user;
+    private String password;
+
+    public PersistenceUnitProperties(Map<String, String> properties) {
+        readPersistenceUnitName(properties);
+        readDataSourceMetaData(properties);
+        readDatabaseNameFromConnectionMetaData();
+        readDialect(properties);
+    }
+
+    private void readPersistenceUnitName(Map<String, String> properties) {
+        persistenceUnitName = Optional.of(properties.get(PERSISTENCE_UNIT_NAME_PROPERTY.value))
+                .orElseThrow(() -> new BibernateException("Persistence name was not provided in properties file"));
+    }
+
+    private void readDialect(Map<String, String> properties) {
+        final String dialectFromProperties = properties.get(DIALECT_PROPERTY.value);
+        if (dialectFromProperties != null) {
+            matchingDialectNames(dialectFromProperties);
```

Comment:
why don't we set dialect from properties in the case when it's defined there?

Reply:
Dialect is not a required property, so we set it from connection metadata. In case when it is provided, we just check for matching dialectName and dbName(from connection metadata). If it's true we set the accorded dialect.